docs: add an aws k8s cluster upgrade guide

consideRatio opened the pull request.

yuvipanda commented on docs/hub-deployment-guide/new-cluster/aws.md:
````diff
@@ -88,19 +101,21 @@ Once you have a `.jsonnet` file, you can render it into a config file that eksct
 
 ```bash
-jsonnet <your-cluster>.jsonnet > <your-cluster>.eksctl.yaml
+jsonnet $CLUSTER_NAME.jsonnet > eksctl-config.yaml
````
I would highly recommend keeping the name of the cluster name in the generated yaml file as well, to prevent accidents where you think you are applying one yaml file but are actually applying a file generated from a different jsonnet! *.eksctl.yaml maybe added to .gitignored if needed.

yuvipanda commented on docs/hub-deployment-guide/upgrade-cluster/aws.md:
```diff
@@ -0,0 +1,208 @@
+(upgrade-cluster:aws)=
```
I would suggest that the upgrade-cluster guide live within the `howto` or `sre-guide` subdirectories

yuvipanda commented on docs/hub-deployment-guide/upgrade-cluster/aws.md:
````diff
@@ -0,0 +1,208 @@
+
+   ```bash
+   # upgrade the kube-proxy daemonset
+   eksctl utils update-kube-proxy --config-file=eksctl-config.yaml --approve
````
Is this specific to any particular set of versions, or required for all? A link to where these commands are sourced from would also be helpful

yuvipanda reviewed (changes requested):
Thanks a lot, Erik! I've left some comments :)

consideRatio commented:
Thanks for the review @yuvipanda!!! I appreciated the ideas in all reivew comments and implemented them all in additional commits.

I've rebased as well as that was relevant to ensure consistent rename of `eksctl-config.yaml` to `$CLUSTER_NAME.eksctl.yaml` from recently merged PRs.

consideRatio commented:
I'll go for a merge here, the comments have been addressed and we can iterate on it as needed - its just docs.
